packages/ripple-nuxt-tide/lib/core/middleware.js

```diff
@@ -285,7 +285,11 @@ export default async function (context, pageData) {
 
     // Landing pages
     if (pageData.tidePage.field_landing_page_c_primary) {
-      addComponentFromPromise(mapping.get(pageData.tidePage.field_landing_page_c_primary), 'appCampaignPrimary')
+      const cPrimaryMapping = await mapping.get(pageData.tidePage.field_landing_page_c_primary)
+      if (!pageData.tidePage.field_show_c_primary_caption) {
+        cPrimaryMapping.data.caption = null
+      }
+      pageData.tidePage['appCampaignPrimary'] = cPrimaryMapping
```

Comment:
Hi @alan-cole Please move this component logic to mapping filters

Reply:
Updated.